File: app/src/main/java/com/walfud/contactsync_android/main/MainView.java
package com.walfud.contactsync_android.main;

import java.util.List;

/**
 * Created by walfud on 2017/4/20.
 */

public interface MainView {
    void show(List<ViewContactData> dataList);

    void loading(boolean show);

    void error(String err);

    class ViewContactData {
        public static final int STATUS_DEFAULT = 0;
        public static final int STATUS_LOCAL_ONLY = 1;
        public static final int STATUS_REMOTE_ONLY = 2;
        public static final int STATUS_CHANGED = 3;

        public String name;
        public List<String> phoneList;
        public int status;
    }
}

File: app/src/main/java/com/walfud/contactsync_android/main/MainView.kt
package com.walfud.contactsync_android.main

/**
 * Created by walfud on 2017/4/20.
 */

interface MainView {
    fun show(dataList: List<ViewContactData>)

    fun loading(show: Boolean)

    fun error(err: String)

    class ViewContactData {

        var name: String? = null
        var phoneList: List<String>? = null
        var status: Int = 0

        companion object {
            val STATUS_DEFAULT = 0
            val STATUS_LOCAL_ONLY = 1
            val STATUS_REMOTE_ONLY = 2
            val STATUS_CHANGED = 3
        }
    }
}
